### PR TITLE
feat(ST-108): fetch classroom for specific student

### PR DESCRIPTION
**Pull request by ghost**
## Description of Change
- Updated the GetClassroomForUser, now it fetches classroom for student also. 

## Associated Ticket Link(s)

- https://trello.com/c/DPrP3vP2

## Screenshots / Screen Recordings
N/A

### CODE REVIEW

File: src/classrooms/classrooms.service.ts [hunk 1]
```diff
@@ -51,20 +52,34 @@ export class ClassroomsService {
...
+            $in: (
+              await this.em.find(Enrollment, { student: user.student })
+            ).map((enrollment) => enrollment.classroom.id),
+          },
+        },
+        {
+          populate: ["teacher", "teacher.user"],
+          orderBy: { classTime: "DESC" },
+        },
+      );
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts]: Let's make this a repository method with an appropriate name. 
Also, I think there is scope to make more readable. There should not be a query inside a clause of another query which is then followed by a .map.

File: src/classrooms/classrooms.service.ts [hunk 2]
```diff
@@ -51,20 +52,34 @@ export class ClassroomsService {
...
+      { populate: ["teacher", "student"] },
     );
-    // TODO: query for fetching classrooms that a student is enrolled in.
+    let classrooms: Classroom[] = [];
+    if (user.student) {
+      classrooms = await this.classroomsRepository.find(
+        {
+          id: {
+            $in: (
+              await this.em.find(Enrollment, { student: user.student })
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts]:
```suggestion
              await this.enrollmentsRepository.find({ student: user.student })
```

File: src/classrooms/classrooms.service.ts [hunk 3]
```diff
@@ -51,20 +53,31 @@ export class ClassroomsService {
...
+      const classroomIds = await this.enrollmentsRepository.getClassroomIdsForStudent(
+        user.student.id,
+      );
+      classrooms = await this.classroomsRepository.find(
+        { id: { $in: classroomIds } },
+        {
+          populate: ["teacher", "teacher.user"],
+          orderBy: { classTime: "DESC" },
+        },
+      );
```
Comment (ishmam-iqbal-sazim) [inline src/classrooms/classrooms.service.ts:71]: [NIT] Does it make sense to abstract this whole snippet into its own method named something like `getClassroomsOfStudent`?

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Left a comment

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**